rotkehlchen/chain/evm/decoding/aave/v3/decoder.py

```diff
@@ -130,3 +145,6 @@ def counterparties() -> tuple[CounterpartyDetails, ...]:
             label=CPT_AAVE_V3.capitalize().replace('-v', ' V'),
             image='aave.svg',
         ),)
+
+    def decoding_by_input_data(self) -> dict[bytes, dict[bytes, Callable]]:
```

Comment:
hmmmm I just noticed something really bad. We don't check for address or list of addresses for the input data rules either. That's bad.

This will hit if any transfer to any contract starts with `b'\xa9\x05\x9c\xbb'`. We need to change this and may as well do it here.

I mean perhaps there is some cases where we need to do input data decoding irrespective of contract. But wealso need one that's limited to specific contracts, like here. Right? Or am I missing something?

And what are you doing here also? Need more of an explanation. What is this input data? Why do you need this rule? What are you trying to accomplish?

If at every transfer of aave you need to check for a mint event shouldn't you just do a post decoding if there is a transfer event of a token that's an aave token? I am not sure what you are trying to do with input data here

Reply:
here we also filter them using the MINT event, that only happens in the aave tokens. So, it is already only running for the transfer of only aave tokens. I did not use post-decoding because there we would need to store a mapping for all the token addresses, and that will occupy a lot of memory.

> And what are you doing here also? Need more of an explanation. What is this input data? Why do you need this rule? What are you trying to accomplish?

here basically check that whenever someone transfers the aToken, do they receive some tokens also. If yes, then we set it's subtype and notes appropriately. I found there is a condition missing here, so I'll add it and also comment the explanation.

Reply:
I don't think you understand my comment. This won't work here.

What if I have a safe? And I transfer through it? Or any other kind of smart contract wallet? This decoding by input data rule is wrong here.

you will need to figure something else out